Prepare v1.2.8 point release

estesp opened the pull request:
Add release notes and update version in preparation for v1.2.8 release
of containerd.

Preview of release notes: https://gist.github.com/estesp/d39a93bf47ed57ba2251265a4b718f25

Signed-off-by: Phil Estes <estesp@linux.vnet.ibm.com>

estesp commented:
Verified travis CI is running Golang 1.12.8:
```
$ go version
go version go1.12.8 linux/amd64
```

estesp commented:
@Random-Liu are there any CRI fixes that need to be vendored into `release/1.2` before we cut a release? Anything to call out in release notes?

Random-Liu commented:
@estesp Yes. Let me do an update.

estesp commented:
Updated notes (https://gist.github.com/estesp/d39a93bf47ed57ba2251265a4b718f25) with CRI additions and rebased

fuweid commented:
ping @estesp I also want to add this issue into release note. https://github.com/containerd/containerd/issues/3481 :)

estesp commented:
Looks like we need to wait on Golang 1.12.9 (which is imminently being released it looks like) to have the [`os.RemoveAll` without `O_CLOEXEC` fix](https://github.com/golang/go/issues/33424) for fd leakage. Others were also [surprised](https://github.com/golang/go/issues/33424#issuecomment-521711404) it wasn't in 1.12.8, but apparently that only took the security fix, and other 1.12.x branch fixes will be in 1.12.9.

fuweid commented:
@estesp yeah. it is painful...

thaJeztah commented on releases/v1.2.8.toml:
```diff
@@ -0,0 +1,37 @@
+### Notable Updates
+* Skip rootfs unmount when no mounts are provided. Fixed by [PR #3148](https://github.com/containerd/containerd/pull/3148) {cherry-picked as [PR #3402](https://github.com/containerd/containerd/pull/3402)}.
+* Close inherited socket file descriptor. Fixed in [PR #3359](https://github.com/containerd/containerd/pull/3359) {cherry-picked as [PR #3364](https://github.com/containerd/containerd/pull/3364)}.
+* Bump libseccomp-golang to v0.9.1. Fixed in [PR #3371](https://github.com/containerd/containerd/pull/3371) {cherry-picked as [PR #3376](https://github.com/containerd/containerd/pull/3376)}.
```
This one was (for now) reverted in https://github.com/containerd/containerd/pull/3538

thaJeztah commented on releases/v1.2.8.toml:
Follow-up for Go 1.12.9; https://github.com/containerd/containerd/pull/3544

(Relates to https://github.com/containerd/containerd/issues/3481, https://github.com/golang/go/issues/33405)

estesp commented:
Updated to 
 - remove mention of libseccomp (change reverted)
 - expand Golang runtime change description to separate 1.12.8 security patches from 1.12.9 bug fix (and added @fuweid requested note of the related issue from @Ace-Tang.
 - clarify opening overview to mention http2 not gRPC for the Golang CVEs

TravisCI is now using 1.12.9 (verified in recent PR run); AppVeyor PR #3544 needs to merge to finalize that across Windows + Linux.

Random-Liu commented on releases/v1.2.8.toml:
```diff
@@ -0,0 +1,39 @@
+The eighth patch release for `containerd` 1.2 provides a series of bug fixes, many
+of them backported from the master branch to correct several known issues around
+manifest lists/indexes and pulling multi-arch, CVEs related to Golang/http2,
+fd leakage in the Golang runtime, and finally mount cleanup related to Cloud Foundry
```
And various fixes for the CRI plugin?

estesp replied:
done

estesp commented:
Updated release notes with general info about CRI having fixes, and pointing to detailed list in next section. Added notes on last few backports (shim hang, and image & process env "merging"), and removed incorrect note on CRI + libseccomp changes, as those were reverted.

estesp commented:
Updated generated release notes gist: https://gist.github.com/estesp/d39a93bf47ed57ba2251265a4b718f25

fuweid commented:
Release Note LGTM 👍 @estesp Thanks!

estesp commented:
@containerd/containerd-maintainers @containerd/containerd-reviewers I believe we are ready to cut the 1.2.8 release; would be good to have some more eyes to make sure the notes captured everything.

crosbymichael commented:
LGTM

Random-Liu commented:
LGTM

dmcgowan commented:
LGTM
